api/envoy/extensions/original_ip_detection/common/v3/original_ip_detection.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+package envoy.extensions.original_ip_detection.common.v3;
+
+import "envoy/type/v3/http_status.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.original_ip_detection.common.v3";
+option java_outer_classname = "OriginalIpDetectionProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Common original IP detection types]
+
+// Common configuration for IP detection extensions.
+//
+// [#next-free-field: 6]
+message OriginalIPDetectionCommonConfig {
```

Comment:
I thought we were getting rid of this since all options don't apply to all extensions?

Reply:
I actually need this for our internal extension and I suspect others might too. But if you think it makes things less clean, I 'll inline it.

Reply:
The key question is whether the defaults are different for different extensions. If it is, it's too confusing IMO. So I would go through and check each field and whether it actually is used by all extensions and has the same default.